src/preferences/broadcastsettings.cpp

```diff
@@ -1,417 +1,135 @@
+    if(!profilesFolder.exists()) {
+        kLogger.warning()
+                << "Profiles folder doesn't exist. Creating it.";
+        profilesFolder.mkpath(profilesFolder.absolutePath());
+    }
 
-QString BroadcastSettings::getStreamDesc() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kStreamDesc),
-            getDefaultStreamDesc());
+    QStringList nameFilters("*.bcp.xml");
+    QFileInfoList files =
+            profilesFolder.entryInfoList(nameFilters, QDir::Files, QDir::Name);
+
+    if(files.count() > 0) {
+        kLogger.warning() << "Found " << files.count() << " profiles.";
+        // Load profiles from filesystem
+        for(QFileInfo fileInfo : files) {
+            BroadcastProfile* profile =
+                    BroadcastProfile::loadFromFile(fileInfo.absoluteFilePath());
+
+            if(profile)
+               m_profiles.push_back(profile);
+        }
+    } else {
+        kLogger.warning()
+                << "No profiles found. Creating default profile.";
+
+        // Create default profile
+        BroadcastProfile* defaultProfile =
+                new BroadcastProfile(kDefaultProfile);
+        // Upgrade from mixxx.cfg format to XML (if required)
+        loadLegacySettings(defaultProfile);
+
+        saveProfile(defaultProfile);
+
+        m_profiles.push_back(defaultProfile);
+        setCurrentProfile(defaultProfile);
+    }
 }
 
-void BroadcastSettings::setStreamDesc(const QString& value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kStreamDesc), value);
-}
+void BroadcastSettings::saveProfile(BroadcastProfile* profile) {
+    if(!profile)
+        return;
 
-QString BroadcastSettings::getDefaultStreamDesc() const {
-    return kDefaultStreamDesc;
+    profile->save(filenameForProfile(profile));
 }
 
-QString BroadcastSettings::getStreamGenre() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kStreamGenre),
-            getDefaultStreamGenre());
+QString BroadcastSettings::filenameForProfile(const QString& profileName) {
+    QString filename = profileName + QString(".bcp.xml");
+    return QDir(getProfilesFolder()).absoluteFilePath(filename);
 }
 
-void BroadcastSettings::setStreamGenre(const QString& value) {
-    m_pConfig->setValue(ConfigKey(kConfigKey, kStreamGenre), value);
-}
+QString BroadcastSettings::filenameForProfile(BroadcastProfile* profile) {
+    if(!profile)
+        return QString();
 
-QString BroadcastSettings::getDefaultStreamGenre() const {
-    return kDefaultStreamGenre;
+    return filenameForProfile(profile->getProfileName());
 }
 
-QString BroadcastSettings::getStreamName() const {
-    return m_pConfig->getValue(
-            ConfigKey(kConfigKey, kStreamName),
-            getDefaultStreamName());
+QString BroadcastSettings::getProfilesFolder() {
+    QString profilesPath(m_pConfig->getSettingsPath());
+    profilesPath.append(QDir::separator() + QString(kProfilesSubfolder));
```

Comment:
Precisely (according to the lines following that error):
- `QTransform operator +(const QTransform &,qreal)`
- `const QSizeF operator +(const QSizeF &,const QSizeF &)`
- `const QSize operator +(const QSize &,const QSize &)`
- `const QPointF operator +(const QPointF &,const QPointF &)`
- `const QPoint operator +(const QPoint &,const QPoint &)`
- `const QString operator +(const QString &,const QByteArray &)`
- `const QString operator +(const QByteArray &,const QString &)`
- `const QString operator +(const QString &,char)`
- `const QString operator +(char,const QString &)`
- `const QString operator +(const char *,const QString &)`
- `const QString operator +(const QString &,const char *)`
- `const QString operator +(QChar,const QString &)`
- `const QString operator +(const QString &,QChar)`
- `const QString operator +(const QString &,const QString &)`
- `const QByteArray operator +(char,const QByteArray &)`
- `const QByteArray operator +(const char *,const QByteArray &)`
- `const QByteArray operator +(const QByteArray &,char)`
- `const QByteArray operator +(const QByteArray &,const char *)`
- `const QByteArray operator +(const QByteArray &,const QByteArray &)'